include/numeric.h

```diff
@@ -11,30 +12,14 @@ namespace dt {
  * http://en.cppreference.com/w/cpp/types/numeric_limits/epsilon
  */
 template<class T>
-typename std::enable_if<std::is_floating_point_v<T>, bool>::type
+typename std::enable_if<std::is_floating_point<T>::value, bool>::type
 almost_equal(T x, T y, int ulp=2)
 {
 	// the machine epsilon has to be scaled to the magnitude of the values used
 	// and multiplied by the desired precision in ULPs (units in the last place)
 	// unless the result is subnormal
-	if constexpr (std::is_same_v<T, float>) {
-		return fabsf(x-y) <= std::numeric_limits<float>::epsilon() * fabsf(x+y) * static_cast<float>(ulp)
-				|| fabsf(x-y) < std::numeric_limits<float>::min();
-	} else if constexpr (std::is_same_v<T, double>) {
-		return fabs(x-y) <= std::numeric_limits<double>::epsilon() * fabs(x+y) * static_cast<double>(ulp)
-				|| fabs(x-y) < std::numeric_limits<double>::min();
-	}
-}
-
-template<typename T>
-T half(const T x)
-{
-	if constexpr (std::is_same_v<T, float>) {
-		return 0.5f * x;
-	} else if constexpr (std::is_same_v<T, double>) {
-		return 0.5 * x;
-	}
-	static_assert(true, "Must be floating-point type");
+	return std::fabs(x-y) <= std::numeric_limits<T>::epsilon() * std::fabs(x+y) * static_cast<T>(ulp)
```

Comment:
I really not a fan of the C++ wrapper for C headers. You can still imitate `if constexpr` with two separate functions with `std::enable_if`

```cpp
template<class T>
typename std::enable_if<std::is_same<T, float>::value, bool>::type
almost_equal(T x, T y, int ulp=2)
{
	return fabsf(x-y) <= std::numeric_limits<float>::epsilon() * fabsf(x+y) * static_cast<float>(ulp)
        	|| fabsf(x-y) < std::numeric_limits<float>::min();
}

template<class T>
typename std::enable_if<std::is_same<T, double>::value, bool>::type
almost_equal(T x, T y, int ulp=2)
{
	return fabs(x-y) <= std::numeric_limits<double>::epsilon() * fabs(x+y) * static_cast<double>(ulp)
	    	|| fabs(x-y) < std::numeric_limits<double>::min();
}
```

Reply:
May I ask why you don't like `cmath`? IMO there should be a really good reason to use a deprecated C header in a C++ project. See for example [here](https://en.cppreference.com/w/cpp/header) - it's clear that `cmath` is the header to use for C++ projects, and `math.h` is considered as deprecated and only there for compatibility reasons. So far I don't see a reason why we should still use `math.h`...

> C++ wrapper for C headers

Whether `cmath` is a wrapper or not is an implementation detail which we should not care about.

Reply:
I had problems with `cmath` on other projects with older compilers. Plus I'm used to do a lot of C and I'd prefer to use the C header.

If you can prove that `cmath` is faster than `math.h` in this case, I'll reconsider using it.

Reply:
> I had problems with cmath on other projects with older compilers.

Maybe you accessed functions *outside* the `std::` namespace? These are not standardized in C++, so for portable code one should only use the functions from within `std::`. With respecting that rule I never had any issues with `cmath`...

Anyway, although I don't agree with using `math.h` at all (look at the complex, unreadable, error-prone, totally useless `enable_if` code bloat), I replaced the code with your suggestion above.

> If you can prove that cmath is faster than math.h in this case, I'll reconsider using it.

I'm pretty sure the performance of both variants is *exactly* the same, so IMHO it would be wasted time to do any performance comparisons...